modules/gltf/gltf_document.cpp

```diff
@@ -3196,10 +3204,60 @@ Error GLTFDocument::_parse_images(Ref<GLTFState> state, const String &p_base_pat
 		// Now we've done our best, fix your scenes.
 		if (img.is_null()) {
 			ERR_PRINT(vformat("glTF: Couldn't load image index '%d' with its given mimetype: %s.", i, mimetype));
+
 			state->images.push_back(Ref<Texture2D>());
+			state->source_images.push_back(Ref<Image>());
 			continue;
 		}
-		state->images.push_back(ImageTexture::create_from_image(img));
+		if (state->handle_binary_image == GLTFState::GLTF_DOCUMENT_DISCARD_TEXTURES) {
+			state->images.push_back(Ref<Texture2D>());
+			state->source_images.push_back(Ref<Image>());
+			continue;
+		} else if (state->handle_binary_image == GLTFState::GLTF_DOCUMENT_EXTRACT_TEXTURES) {
+			String extracted_image_name = image_name.get_file().get_basename();
+			img->set_name(extracted_image_name);
+			if (state->base_path.is_empty()) {
+				state->images.push_back(Ref<Texture2D>());
+				state->source_images.push_back(Ref<Image>());
+			} else if (img->get_name().is_empty()) {
+				WARN_PRINT(vformat("glTF: Image index '%d' couldn't be named. Skipping it.", i));
+				state->images.push_back(Ref<Texture2D>());
+				state->source_images.push_back(Ref<Image>());
+			} else {
+				String file_path = state->get_base_path() + state->filename.get_basename() + "_" + img->get_name() + ".png";
+				img->save_png(file_path);
+				Ref<Texture2D> saved_image = ResourceLoader::load(file_path, "Texture2D");
```

Comment:
I wasn't aware that you can load a Texture2D from a newly-written path without going through the importer pipeline (e.g. calling reimport_files() or another API). Does this load the texture uncompressed?

Reply:
This part is broken. Stuck.

Reply:
reduz noted in RocketChat to look at how the lightmapper does this. Lightmaps use ResourceLoader::import() followed by ResourceLoader::load().